kyo-core/shared/src/test/scala/kyoTest/envsTest.scala

```diff
@@ -28,7 +29,7 @@ class envsTest extends KyoTest:
         trait Super:
             def i = 42
         case class Sub() extends Super
-        assert(Envs.run[Super](Sub())(Envs.use[Super](_.i)).pure == 42)
```

Comment:
Is this change still necessary after adding `>: Nothing` in `run`?

Reply:
Oh. This wasn't due to type inference reasons. It's just that we don't need to upcast, because the type map will provide Subtypes to Supertypes automatically.

Reply:
And I inlined the `RunDsl` 😜 because I always find that pattern of allowing partially applied type parameters to be pretty confusing (esp. to newcomers command-clicking through the codebase). But I'm happy to revert if you'd prefer!

Reply:
makes sense, thanks for the clarification! Yep, I'm not a fan of the `*Dsl` encoding, hopefully we'll eventually be able to use [clause interleaving](https://docs.scala-lang.org/sips/clause-interleaving.html) instead but I think it's still marked as experimental.

Reply:
Oh, awesome! I'm glad they're adding that.